src/tape/ensemble.py

```diff
@@ -1110,6 +1118,16 @@ def s2n_inter_quartile_range(flux, err):
         if isinstance(on, str):
             on = [on]  # Convert to list if only one column is passed
 
+        if by_band:
+            if self._band_col not in on:
+                on += [self._band_col]
+            elif (
+                on[-1] != self._band_col
+            ):  # Unsure if this is neccesary but it seems likely to produce wrong behavior
```

Comment:
Can we always ensure correct `on` ordering for the user? We're doing a linear search of `on` anyway and this would prevent an error and let us remove a weird caveat from the docstring.

We also 

I think the code could look something like

```
if by_band:
    if self._band_col not in on:
        on += [self._band_col]
    elif on[-1] != self._band_col:
        # Ensure band is the final column in the `on` list
        on[on.index(self._band_col)] = on[-1]
        on[-1] = self._band_col
```

Alternatively could use `list.index()` to get the index and check if present in `on` in one pass, but would need a try-except for handling if it's missing from the list (which would replace the if-elif here)

Reply:
This sounds reasonable, will try it out!

Reply:
Implemented in latest commit

Reply:
Thank you!